--- ./libs/libmyth/mythcontext.cpp~	2005-04-11 21:59:29.000000000 +0300
+++ ./libs/libmyth/mythcontext.cpp	2005-09-05 20:20:24.000000000 +0300
@@ -874,7 +878,7 @@
 
 QString MythContext::GetLibraryDir(void) 
 { 
-    return d->m_installprefix + "/lib/mythtv/"; 
+    return d->m_installprefix + "/lib64/mythtv/"; 
 }
 
 QString MythContext::GetThemesParentDir(void) 
diff -ur mythtv-lib32/filters/filter-common.pro mythtv/filters/filter-common.pro
--- mythtv-lib32/filters/filter-common.pro	2005-10-18 02:33:41.000000000 +0300
+++ mythtv/filters/filter-common.pro	2005-10-19 20:56:16.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 CONFIG -= moc qt
 CONFIG += plugin thread
-target.path = $${PREFIX}/lib/mythtv/filters
+target.path = $${PREFIX}/lib64/mythtv/filters
 INSTALLS = target
 
 QMAKE_CFLAGS_RELEASE += -Wno-missing-prototypes
diff -ur mythtv-lib32/libs/libavcodec/libavcodec.pro mythtv/libs/libavcodec/libavcodec.pro
--- mythtv-lib32/libs/libavcodec/libavcodec.pro	2005-10-18 02:33:32.000000000 +0300
+++ mythtv/libs/libavcodec/libavcodec.pro	2005-10-19 20:56:16.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 TARGET = mythavcodec-$$LIBVERSION
 CONFIG += thread dll warn_off
-target.path = $${PREFIX}/lib
+target.path = $${PREFIX}/lib64
 INSTALLS = target
 
 VERSION = 0.18.0
diff -ur mythtv-lib32/libs/libavformat/libavformat.pro mythtv/libs/libavformat/libavformat.pro
--- mythtv-lib32/libs/libavformat/libavformat.pro	2005-10-18 02:33:08.000000000 +0300
+++ mythtv/libs/libavformat/libavformat.pro	2005-10-19 20:56:16.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 TARGET = mythavformat-$$LIBVERSION
 CONFIG += thread dll warn_off
-target.path = $${PREFIX}/lib
+target.path = $${PREFIX}/lib64
 INSTALLS = target
 
 VERSION = 0.18.0 
diff -ur mythtv-lib32/libs/libavutil/libavutil.pro mythtv/libs/libavutil/libavutil.pro
--- mythtv-lib32/libs/libavutil/libavutil.pro	2005-10-18 02:33:33.000000000 +0300
+++ mythtv/libs/libavutil/libavutil.pro	2005-10-19 20:56:16.000000000 +0300
@@ -5,7 +5,7 @@
 TARGET = mythavutil-$$LIBVERSION
 CONFIG += thread dll warn_off
 CONFIG -= qt
-target.path = $${PREFIX}/lib
+target.path = $${PREFIX}/lib64
 INSTALLS = target
 
 VERSION = 0.18.0
diff -ur mythtv-lib32/libs/libmyth/libmyth.pro mythtv/libs/libmyth/libmyth.pro
--- mythtv-lib32/libs/libmyth/libmyth.pro	2005-10-18 02:33:16.000000000 +0300
+++ mythtv/libs/libmyth/libmyth.pro	2005-10-19 20:56:16.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 TARGET = myth-$$LIBVERSION
 CONFIG += thread dll
-target.path = $${PREFIX}/lib
+target.path = $${PREFIX}/lib64
 INSTALLS = target
 
 VERSION = 0.18.0
diff -ur mythtv-lib32/libs/libmythmpeg2/libmythmpeg2.pro mythtv/libs/libmythmpeg2/libmythmpeg2.pro
--- mythtv-lib32/libs/libmythmpeg2/libmythmpeg2.pro	2005-10-18 02:33:03.000000000 +0300
+++ mythtv/libs/libmythmpeg2/libmythmpeg2.pro	2005-10-19 20:56:16.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 TARGET = mythmpeg2-$$LIBVERSION
 CONFIG += thread staticlib warn_off
-target.path = $${PREFIX}/lib
+target.path = $${PREFIX}/lib64
 INSTALLS = target
 
 VERSION = 0.18.0 
diff -ur mythtv-lib32/libs/libmythsamplerate/libmythsamplerate.pro mythtv/libs/libmythsamplerate/libmythsamplerate.pro
--- mythtv-lib32/libs/libmythsamplerate/libmythsamplerate.pro	2005-10-18 02:33:12.000000000 +0300
+++ mythtv/libs/libmythsamplerate/libmythsamplerate.pro	2005-10-19 20:56:16.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 TARGET = mythsamplerate-$$LIBVERSION
 CONFIG += thread staticlib warn_off
-target.path = $${PREFIX}/lib
+target.path = $${PREFIX}/lib64
 INSTALLS = target
 
 VERSION = 0.18.0 
diff -ur mythtv-lib32/libs/libmythsoundtouch/libmythsoundtouch.pro mythtv/libs/libmythsoundtouch/libmythsoundtouch.pro
--- mythtv-lib32/libs/libmythsoundtouch/libmythsoundtouch.pro	2005-10-18 02:33:03.000000000 +0300
+++ mythtv/libs/libmythsoundtouch/libmythsoundtouch.pro	2005-10-19 20:56:16.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 TARGET = mythsoundtouch-$$LIBVERSION
 CONFIG += thread staticlib warn_off
-target.path = $${PREFIX}/lib
+target.path = $${PREFIX}/lib64
 INSTALLS = target
 
 VERSION = 0.18.0 
diff -ur mythtv-lib32/libs/libmythtv/libmythtv.pro mythtv/libs/libmythtv/libmythtv.pro
--- mythtv-lib32/libs/libmythtv/libmythtv.pro	2005-10-18 02:33:02.000000000 +0300
+++ mythtv/libs/libmythtv/libmythtv.pro	2005-10-19 20:56:16.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 TARGET = mythtv-$$LIBVERSION
 CONFIG += thread dll
-target.path = $${PREFIX}/lib
+target.path = $${PREFIX}/lib64
 INSTALLS = target
 
 VERSION = 0.18.0
diff -ur mythtv-lib32/libs/libmythui/libmythui.pro mythtv/libs/libmythui/libmythui.pro
--- mythtv-lib32/libs/libmythui/libmythui.pro	2005-10-18 02:33:13.000000000 +0300
+++ mythtv/libs/libmythui/libmythui.pro	2005-10-19 20:56:16.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 TARGET = mythui-$$LIBVERSION
 CONFIG += debug thread dll 
-target.path = $${PREFIX}/lib
+target.path = $${PREFIX}/lib64
 INSTALLS = target
 
 INCLUDEPATH += ../libmyth
