--- configure.orig	2008-11-01 18:37:08.000000000 +0100
+++ configure	2008-11-01 18:42:53.000000000 +0100
@@ -1279,8 +1279,8 @@
     arch=`arch`
     if test x"$arch" = x"ppc" -o x"$arch" = x"ppc64" ; then
         processor=`/usr/bin/hostinfo | grep "Processor type:"`
-        processor=${processor#Processor type: *}
-        processor=${processor% (*}
+        processor="${processor#Processor type:\ *}"
+        processor="${processor% \(*}"
         tune=$processor
     else
             # 10.4 only:
