Index: configure
===================================================================
--- configure	(revision 9014)
+++ configure	(working copy)
@@ -1859,8 +1859,7 @@
 audio_arts=no
 if (artsc-config --version) > /dev/null 2>&1; then
 audio_arts_libs=`artsc-config --libs`
-audio_arts_tmp=`artsc-config --cflags`
-audio_arts_cflags=`echo $audio_arts_tmp | sed "s/include\/artsc/include/g"`
+audio_arts_cflags=`artsc-config --cflags`
 if $cc -o $TMPE $TMPC $audio_arts_cflags $audio_arts_libs > /dev/null 2>&1; then
 audio_arts=yes
 fi
