--- mythtv/libs/libmythtv/jobqueue.cpp~	2006-09-05 04:56:41.000000000 +0300
+++ mythtv/libs/libmythtv/jobqueue.cpp	2006-09-05 04:57:45.000000000 +0300
@@ -1955,7 +1955,7 @@
 
     if (runningJobCommands[key] == "mythcommflag")
     {
-        path = gContext->GetInstallPrefix() + "/bin/mythcommflag";
+        path = gContext->GetInstallPrefix() + "/sbin/mythcommflag";
         command = QString("%1 -j %2 -V %3")
                           .arg(path).arg(jobID).arg(print_verbose_messages);
     }
--- mythtv/programs/mythcommflag/mythcommflag.pro~	2005-10-18 02:33:37.000000000 +0300
+++ mythtv/programs/mythcommflag/mythcommflag.pro	2006-02-05 16:20:01.000000000 +0200
@@ -4,7 +4,7 @@
 
 TEMPLATE = app
 CONFIG += thread
-target.path = $${PREFIX}/bin
+target.path = $${PREFIX}/sbin
 INSTALLS = target
 
 QMAKE_CLEAN += $(TARGET)
--- mythtv/programs/mythbackend/mythbackend.pro~	2006-01-29 18:32:00.000000000 +0200
+++ mythtv/programs/mythbackend/mythbackend.pro	2006-02-05 16:20:52.000000000 +0200
@@ -6,7 +6,7 @@
 TEMPLATE = app
 CONFIG += thread
 TARGET = mythbackend
-target.path = $${PREFIX}/bin
+target.path = $${PREFIX}/sbin
 INSTALLS = target
 
 QMAKE_CLEAN += $(TARGET)
--- mythtv/programs/mythjobqueue/mythjobqueue.pro~	2005-10-18 02:33:36.000000000 +0300
+++ mythtv/programs/mythjobqueue/mythjobqueue.pro	2006-02-05 16:21:08.000000000 +0200
@@ -4,7 +4,7 @@
 
 TEMPLATE = app
 CONFIG += thread
-target.path = $${PREFIX}/bin
+target.path = $${PREFIX}/sbin
 INSTALLS = target
 
 QMAKE_CLEAN += $(TARGET)
--- mythtv/programs/mythlcdserver/mythlcdserver.pro~	2005-10-18 02:33:38.000000000 +0300
+++ mythtv/programs/mythlcdserver/mythlcdserver.pro	2006-02-05 16:21:25.000000000 +0200
@@ -5,7 +5,7 @@
 TEMPLATE = app
 CONFIG += thread
 TARGET = mythlcdserver
-target.path = $${PREFIX}/bin
+target.path = $${PREFIX}/sbin
 
 INSTALLS += target
 
