--- ./libs/libmythsamplerate/libmythsamplerate.pro~	2005-05-06 23:19:54.000000000 +0300
+++ ./libs/libmythsamplerate/libmythsamplerate.pro	2005-09-04 00:10:56.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 TARGET = mythsamplerate-$$LIBVERSION
 CONFIG += thread staticlib warn_off
-target.path = $${PREFIX}/lib
+target.path = $${LIBDIR}
 INSTALLS = target
 
 VERSION = 0.18.1 
diff -ur mythtv-0.18.1/filters/adjust/adjust.pro mythtv-0.18.1.fixed/filters/adjust/adjust.pro
--- mythtv-0.18.1/filters/adjust/adjust.pro	2005-01-29 07:21:36.000000000 +0200
+++ mythtv-0.18.1.fixed/filters/adjust/adjust.pro	2005-09-04 00:25:07.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 CONFIG -= moc qt
 CONFIG += plugin thread
-target.path = $${PREFIX}/lib/mythtv/filters
+target.path = $${LIBDIR}/mythtv/filters
 INSTALLS = target
 
 INCLUDEPATH += ../../libs/libmythtv
diff -ur mythtv-0.18.1/filters/bobdeint/bobdeint.pro mythtv-0.18.1.fixed/filters/bobdeint/bobdeint.pro
--- mythtv-0.18.1/filters/bobdeint/bobdeint.pro	2005-01-29 07:21:36.000000000 +0200
+++ mythtv-0.18.1.fixed/filters/bobdeint/bobdeint.pro	2005-09-04 00:25:16.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 CONFIG -= moc qt
 CONFIG += plugin thread
-target.path = $${PREFIX}/lib/mythtv/filters
+target.path = $${LIBDIR}/mythtv/filters
 INSTALLS = target
 
 INCLUDEPATH += ../../libs/libmythtv
diff -ur mythtv-0.18.1/filters/convert/convert.pro mythtv-0.18.1.fixed/filters/convert/convert.pro
--- mythtv-0.18.1/filters/convert/convert.pro	2005-01-29 07:21:36.000000000 +0200
+++ mythtv-0.18.1.fixed/filters/convert/convert.pro	2005-09-04 00:24:55.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 CONFIG -= moc qt
 CONFIG += plugin thread
-target.path = $${PREFIX}/lib/mythtv/filters
+target.path = $${LIBDIR}/mythtv/filters
 INSTALLS = target
 
 INCLUDEPATH += ../../libs/libmythtv
diff -ur mythtv-0.18.1/filters/crop/crop.pro mythtv-0.18.1.fixed/filters/crop/crop.pro
--- mythtv-0.18.1/filters/crop/crop.pro	2005-01-29 07:21:36.000000000 +0200
+++ mythtv-0.18.1.fixed/filters/crop/crop.pro	2005-09-04 00:24:52.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 CONFIG -= moc qt
 CONFIG += plugin thread
-target.path = $${PREFIX}/lib/mythtv/filters
+target.path = $${LIBDIR}/mythtv/filters
 INSTALLS = target
 
 INCLUDEPATH += ../../libs/libmythtv
diff -ur mythtv-0.18.1/filters/denoise3d/denoise3d.pro mythtv-0.18.1.fixed/filters/denoise3d/denoise3d.pro
--- mythtv-0.18.1/filters/denoise3d/denoise3d.pro	2005-01-29 07:21:36.000000000 +0200
+++ mythtv-0.18.1.fixed/filters/denoise3d/denoise3d.pro	2005-09-04 00:25:21.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 CONFIG -= moc qt
 CONFIG += plugin thread
-target.path = $${PREFIX}/lib/mythtv/filters
+target.path = $${LIBDIR}/mythtv/filters
 INSTALLS = target
 
 INCLUDEPATH += ../../libs/libmythtv
diff -ur mythtv-0.18.1/filters/force/force.pro mythtv-0.18.1.fixed/filters/force/force.pro
--- mythtv-0.18.1/filters/force/force.pro	2005-01-29 07:21:36.000000000 +0200
+++ mythtv-0.18.1.fixed/filters/force/force.pro	2005-09-04 00:24:57.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 CONFIG -= moc qt
 CONFIG += plugin thread
-target.path = $${PREFIX}/lib/mythtv/filters
+target.path = $${LIBDIR}/mythtv/filters
 INSTALLS = target
 
 INCLUDEPATH += ../../libs/libmythtv
diff -ur mythtv-0.18.1/filters/invert/invert.pro mythtv-0.18.1.fixed/filters/invert/invert.pro
--- mythtv-0.18.1/filters/invert/invert.pro	2005-01-29 07:21:36.000000000 +0200
+++ mythtv-0.18.1.fixed/filters/invert/invert.pro	2005-09-04 00:25:09.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 CONFIG -= moc qt
 CONFIG += plugin thread
-target.path = $${PREFIX}/lib/mythtv/filters
+target.path = $${LIBDIR}/mythtv/filters
 INSTALLS = target
 
 INCLUDEPATH += ../../libs/libmythtv
diff -ur mythtv-0.18.1/filters/kerneldeint/kerneldeint.pro mythtv-0.18.1.fixed/filters/kerneldeint/kerneldeint.pro
--- mythtv-0.18.1/filters/kerneldeint/kerneldeint.pro	2005-01-29 07:21:36.000000000 +0200
+++ mythtv-0.18.1.fixed/filters/kerneldeint/kerneldeint.pro	2005-09-04 00:25:14.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 CONFIG -= moc qt
 CONFIG += plugin thread
-target.path = $${PREFIX}/lib/mythtv/filters
+target.path = $${LIBDIR}/mythtv/filters
 INSTALLS = target
 
 INCLUDEPATH += ../../libs/libmythtv
diff -ur mythtv-0.18.1/filters/linearblend/linearblend.pro mythtv-0.18.1.fixed/filters/linearblend/linearblend.pro
--- mythtv-0.18.1/filters/linearblend/linearblend.pro	2005-01-29 07:21:36.000000000 +0200
+++ mythtv-0.18.1.fixed/filters/linearblend/linearblend.pro	2005-09-04 00:24:59.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 CONFIG -= moc qt
 CONFIG += plugin thread
-target.path = $${PREFIX}/lib/mythtv/filters
+target.path = $${LIBDIR}/mythtv/filters
 INSTALLS = target
 
 INCLUDEPATH += ../../libs/libmythtv
diff -ur mythtv-0.18.1/filters/onefield/onefield.pro mythtv-0.18.1.fixed/filters/onefield/onefield.pro
--- mythtv-0.18.1/filters/onefield/onefield.pro	2005-01-29 07:21:36.000000000 +0200
+++ mythtv-0.18.1.fixed/filters/onefield/onefield.pro	2005-09-04 00:25:02.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 CONFIG -= moc qt
 CONFIG += plugin thread
-target.path = $${PREFIX}/lib/mythtv/filters
+target.path = $${LIBDIR}/mythtv/filters
 INSTALLS = target
 
 INCLUDEPATH += ../../libs/libmythtv
diff -ur mythtv-0.18.1/filters/postprocess/postprocess.pro mythtv-0.18.1.fixed/filters/postprocess/postprocess.pro
--- mythtv-0.18.1/filters/postprocess/postprocess.pro	2005-01-29 07:21:36.000000000 +0200
+++ mythtv-0.18.1.fixed/filters/postprocess/postprocess.pro	2005-09-04 00:25:12.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 CONFIG -= moc qt
 CONFIG += plugin thread
-target.path = $${PREFIX}/lib/mythtv/filters
+target.path = $${LIBDIR}/mythtv/filters
 INSTALLS = target
 
 INCLUDEPATH += ../../libs/libmythtv ../../libs/libavcodec
diff -ur mythtv-0.18.1/filters/quickdnr/quickdnr.pro mythtv-0.18.1.fixed/filters/quickdnr/quickdnr.pro
--- mythtv-0.18.1/filters/quickdnr/quickdnr.pro	2005-01-29 07:21:36.000000000 +0200
+++ mythtv-0.18.1.fixed/filters/quickdnr/quickdnr.pro	2005-09-04 00:25:18.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 CONFIG -= moc qt
 CONFIG += plugin thread
-target.path = $${PREFIX}/lib/mythtv/filters
+target.path = $${LIBDIR}/mythtv/filters
 INSTALLS = target
 
 INCLUDEPATH += ../../libs/libmythtv
diff -ur mythtv-0.18.1/libs/libavcodec/libavcodec.pro mythtv-0.18.1.fixed/libs/libavcodec/libavcodec.pro
--- mythtv-0.18.1/libs/libavcodec/libavcodec.pro	2005-05-06 23:19:54.000000000 +0300
+++ mythtv-0.18.1.fixed/libs/libavcodec/libavcodec.pro	2005-09-04 00:24:44.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 TARGET = mythavcodec-$$LIBVERSION
 CONFIG += thread dll warn_off
-target.path = $${PREFIX}/lib
+target.path = $${LIBDIR}
 INSTALLS = target
 
 VERSION = 0.18.1
diff -ur mythtv-0.18.1/libs/libavformat/libavformat.pro mythtv-0.18.1.fixed/libs/libavformat/libavformat.pro
--- mythtv-0.18.1/libs/libavformat/libavformat.pro	2005-05-06 23:19:54.000000000 +0300
+++ mythtv-0.18.1.fixed/libs/libavformat/libavformat.pro	2005-09-04 00:24:47.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 TARGET = mythavformat-$$LIBVERSION
 CONFIG += thread dll warn_off
-target.path = $${PREFIX}/lib
+target.path = $${LIBDIR}
 INSTALLS = target
 
 VERSION = 0.18.1 
diff -ur mythtv-0.18.1/libs/libmyth/libmyth.pro mythtv-0.18.1.fixed/libs/libmyth/libmyth.pro
--- mythtv-0.18.1/libs/libmyth/libmyth.pro	2005-05-06 23:19:54.000000000 +0300
+++ mythtv-0.18.1.fixed/libs/libmyth/libmyth.pro	2005-09-04 00:24:37.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 TARGET = myth-$$LIBVERSION
 CONFIG += thread dll
-target.path = $${PREFIX}/lib
+target.path = $${LIBDIR}
 INSTALLS = target
 
 VERSION = 0.18.1
diff -ur mythtv-0.18.1/libs/libmythmpeg2/libmythmpeg2.pro mythtv-0.18.1.fixed/libs/libmythmpeg2/libmythmpeg2.pro
--- mythtv-0.18.1/libs/libmythmpeg2/libmythmpeg2.pro	2005-05-06 23:19:54.000000000 +0300
+++ mythtv-0.18.1.fixed/libs/libmythmpeg2/libmythmpeg2.pro	2005-09-04 00:24:41.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 TARGET = mythmpeg2-$$LIBVERSION
 CONFIG += thread staticlib warn_off
-target.path = $${PREFIX}/lib
+target.path = $${LIBDIR}
 INSTALLS = target
 
 VERSION = 0.18.1 
diff -ur mythtv-0.18.1/libs/libmythsoundtouch/libmythsoundtouch.pro mythtv-0.18.1.fixed/libs/libmythsoundtouch/libmythsoundtouch.pro
--- mythtv-0.18.1/libs/libmythsoundtouch/libmythsoundtouch.pro	2005-05-06 23:19:54.000000000 +0300
+++ mythtv-0.18.1.fixed/libs/libmythsoundtouch/libmythsoundtouch.pro	2005-09-04 00:24:50.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 TARGET = mythsoundtouch-$$LIBVERSION
 CONFIG += thread staticlib warn_off
-target.path = $${PREFIX}/lib
+target.path = $${LIBDIR}
 INSTALLS = target
 
 VERSION = 0.18.1 
diff -ur mythtv-0.18.1/libs/libmythtv/libmythtv.pro mythtv-0.18.1.fixed/libs/libmythtv/libmythtv.pro
--- mythtv-0.18.1/libs/libmythtv/libmythtv.pro	2005-05-06 23:19:54.000000000 +0300
+++ mythtv-0.18.1.fixed/libs/libmythtv/libmythtv.pro	2005-09-04 00:24:34.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 TARGET = mythtv-$$LIBVERSION
 CONFIG += thread dll
-target.path = $${PREFIX}/lib
+target.path = $${LIBDIR}
 INSTALLS = target
 
 VERSION = 0.18.1
diff -ur mythtv-0.18.1/libs/libmythui/libmythui.pro mythtv-0.18.1.fixed/libs/libmythui/libmythui.pro
--- mythtv-0.18.1/libs/libmythui/libmythui.pro	2005-05-06 23:19:54.000000000 +0300
+++ mythtv-0.18.1.fixed/libs/libmythui/libmythui.pro	2005-09-04 00:24:32.000000000 +0300
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 TARGET = mythui-$$LIBVERSION
 CONFIG += debug thread dll 
-target.path = $${PREFIX}/lib
+target.path = $${LIBDIR}
 INSTALLS = target
 INCLUDEPATH += ../../libs/libmyth
 INCLUDEPATH += ../..
