--- ./configure~	2005-10-19 20:39:45.000000000 +0300
+++ ./configure	2005-10-19 20:54:02.000000000 +0300
@@ -265,7 +265,7 @@
   ;;
   x86_64)
     cpu="x86"
-    canon_arch="`cc -dumpmachine | sed -e 's,\([^-]*\)-.*,\1,'`"
+    canon_arch="`$cc -dumpmachine | sed -e 's,\([^-]*\)-.*,\1,'`"
     if [ x"$canon_arch" = x"x86_64" -o x"$canon_arch" = x"amd64" ]; then
       if [ -z "`echo $CFLAGS | grep -- -m32`"  ]; then
         cpu="x86_64"
