diff -urN mythtv-0.20/libs/libmythdvdnav/libmythdvdnav.pro mythtv-0.20.new/libs/libmythdvdnav/libmythdvdnav.pro
--- mythtv-0.20/libs/libmythdvdnav/libmythdvdnav.pro	2006-02-17 11:27:03.000000000 +0100
+++ mythtv-0.20.new/libs/libmythdvdnav/libmythdvdnav.pro	2006-10-06 09:12:47.000000000 +0200
@@ -3,7 +3,7 @@
 
 TEMPLATE = lib
 TARGET = mythdvdnav-$$LIBVERSION
-CONFIG += thread staticlib warn_off
+CONFIG += thread dll warn_off
 target.path = $${LIBDIR}
 
 INCLUDEPATH += ../../
diff -urN mythtv-0.20/programs/programs-libs.pro mythtv-0.20.new/programs/programs-libs.pro
--- mythtv-0.20/programs/programs-libs.pro	2006-07-09 01:44:07.000000000 +0200
+++ mythtv-0.20.new/programs/programs-libs.pro	2006-10-06 10:03:45.000000000 +0200
@@ -1,16 +1,16 @@
-INCLUDEPATH += ../.. ../../libs/ ../../libs/libmyth ../../libs/libmythtv
+INCLUDEPATH += ../.. ../../libs/ ../../libs/libmythdvdnav ../../libs/libmyth ../../libs/libmythtv
 INCLUDEPATH += ../../libs/libavutil ../../libs/libavformat ../../libs/libavcodec
 INCLUDEPATH += ../../libs/libmythupnp ../../libs/libmythui
 INCLUDEPATH += ../../libs/libmythlivemedia
 
-LIBS += -L../../libs/libmyth -L../../libs/libmythtv
+LIBS += -L../../libs/libmythdvdnav -L../../libs/libmyth -L../../libs/libmythtv
 LIBS += -L../../libs/libavutil -L../../libs/libavcodec -L../../libs/libavformat
 LIBS += -L../../libs/libmythfreemheg
 LIBS += -L../../libs/libmythui
 LIBS += -L../../libs/libmythupnp
 LIBS += -L../../libs/libmythlivemedia
 
-LIBS += -lmythtv-$$LIBVERSION -lmythavformat-$$LIBVERSION
+LIBS += -lmythdvdnav-$$LIBVERSION -lmythtv-$$LIBVERSION -lmythavformat-$$LIBVERSION
 LIBS += -lmythavutil-$$LIBVERSION -lmythavcodec-$$LIBVERSION 
 LIBS += -lmythfreemheg-$$LIBVERSION
 LIBS += -lmythupnp-$$LIBVERSION 
@@ -32,7 +33,7 @@
 TARGETDEPS += ../../libs/libmythupnp/libmythupnp-$${LIBVERSION}.$${QMAKE_EXTENSION_SHLIB}
 TARGETDEPS += ../../libs/libmythlivemedia/libmythlivemedia-$${LIBVERSION}.$${QMAKE_EXTENSION_SHLIB}
 
-DEPENDPATH += ../.. ../../libs ../../libs/libmyth ../../libs/libmythtv
+DEPENDPATH += ../.. ../../libs ../../libs/libmyth ../../libs/libmythdvdnav ../../libs/libmythtv
 DEPENDPATH += ../../libs/libavutil ../../libs/libavformat ../../libs/libsavcodec
 DEPENDPATH += ../../libs/libmythupnp ../../libs/libmythui
 DEPENDPATH += ../../libs/libmythlivemedia
