diff -urN mythtv-0.22rc1.orig/configure mythtv-0.22rc1/configure
--- mythtv-0.22rc1.orig/configure	2009-10-09 00:30:46.000000000 +0200
+++ mythtv-0.22rc1/configure	2009-10-17 23:10:29.000000000 +0200
@@ -1564,8 +1564,8 @@
     arch=`arch`
     if test x"$arch" = x"ppc" -o x"$arch" = x"ppc64" ; then
         processor=`/usr/bin/hostinfo | grep "Processor type:"`
-        processor=${processor#Processor type: *}
-        processor=${processor% (*}
+        processor="${processor#Processor type:\ *}"
+        processor="${processor% \(*}"
         tune=$processor
     else
             # 10.4 only:
