---
 libs/libmythtv/tv_rec.h           |    2 	1 +	1 -	0 !
 5 files changed, 6 insertions(+), 6 deletions(-)

Index: mythtv/libs/libmythtv/tv_rec.h
===================================================================
--- mythtv.orig/libs/libmythtv/tv_rec.h	2010-05-16 14:58:08.000000000 -0400
+++ mythtv/libs/libmythtv/tv_rec.h	2010-07-01 22:34:54.000000000 -0400
@@ -93,7 +93,7 @@
 class TuningRequest
 {
   public:
-    TuningRequest(uint f) :
+    TuningRequest(uint f = 0) :
         flags(f), program(NULL), channel(QString::null),
         input(QString::null), majorChan(0), minorChan(0), progNum(-1) {;}
     TuningRequest(uint f, RecordingInfo *p) :
